errors.go

```diff
@@ -0,0 +1,58 @@
+package quic
+
+import (
+	"fmt"
+
+	"github.com/lucas-clemente/quic-go/internal/qerr"
+)
+
+type (
+	TransportError          = qerr.TransportError
```

Comment:
Now this is a bit weird. We have a
1. `TransportError`: the type of a transport error
2. `TransportErrorCode`: the type of the error code for transport errors (a `uint64`)
3. `TransportErrorErrorCode`: the error code the the TRANSPORT_ERROR (see `0xa` in https://www.ietf.org/archive/id/draft-ietf-quic-transport-34.html#name-transport-error-codes)

Reply:
>     2\. `TransportErrorCode`

I guess just `ErrorCode` is not an option?



> `TransportErrorErrorCode`: the error code the the TRANSPORT_ERROR (see `0xa` in https://www.ietf.org/archive/id/draft-ietf-quic-transport-34.html#name-transport-error-codes)

not sure I follow; that page shows PROTOCOL_VIOLATION for 0xa. Could you not use ProtocolViolation?

Reply:
Sorry, I meant 0xc.